src/Features/LanguageServer/Protocol/Api/InProcLanguageServer.cs

```diff
@@ -0,0 +1,404 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.ErrorReporting;
+using Microsoft.CodeAnalysis.LanguageServer.Handler;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Roslyn.Utilities;
+using StreamJsonRpc;
+
+using LSP = Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Api
+{
+    internal class InProcLanguageServer : IAsyncDisposable
```

Comment:
The entirety of this was just moved from the old InProcLanguageServer class in the VS layer to the LSP project.  I removed VS LSP extension methods (and VS LSP extension types) from this as well as LSP publish diagnostics support (pull diagnostics will be in LSP 3.17 spec so I'd rather not keep around the old really hacky implementation of LSP publish).

Everything else is essentially the same.